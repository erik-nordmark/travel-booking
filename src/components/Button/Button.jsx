import React from 'react';
import styles from './Button.module.scss';

const Colors = {
	DARK: Symbol('dark'),
	LIGHT: Symbol('light'),
};

const Button = ({ label, color, full, onClick, disabled }) => {
	return (
		<button
			type='button'
			disabled={disabled}
			data-testid='button'
			onClick={onClick}
			style={{ width: full ? '100%' : '' }}
			className={`${styles.button} ${
				color === Colors.DARK ? styles.dark : styles.light
			}`}
		>
			{label}
		</button>
	);
};

export const DarkButton = ({ label, full, onClick, disabled }) => {
	return (
		<Button
			label={label}
			color={Colors.DARK}
			full={full}
			onClick={onClick}
			disabled={disabled}
		/>
	);
};

export const LightButton = ({ label, full, onClick, disabled }) => {
	return (
		<Button
			label={label}
			color={Colors.LIGHT}
			full={full}
			onClick={onClick}
			disabled={disabled}
		/>
	);
};
