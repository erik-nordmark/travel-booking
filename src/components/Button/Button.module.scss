a {
	text-decoration: none;
}

button:disabled,
button[disabled] {
	opacity: 0.2;
}

.button {
	width: 225px;
	height: 55px;
	display: flex;
	justify-content: center;
	align-items: center;
	font-family: 'Fahkwang', sans-serif;
	font-style: normal;
	font-weight: 600;
	font-size: 14px;
	line-height: 18px;
	cursor: pointer;
}

.dark {
	background: #000000;
	color: #ffffff;
}

.light {
	background: #ffffff;
	color: #000000;
}

@media screen and (max-width: 800px) {
	.button {
		width: 100%;
	}
}
