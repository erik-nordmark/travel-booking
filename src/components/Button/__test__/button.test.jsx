import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import { DarkButton } from 'components/Button/Button';

test('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<DarkButton />, div);
});

test('renders button correctly', () => {
	const { getByTestId } = render(<DarkButton label='test' disabled full />);
	expect(getByTestId('button')).toHaveTextContent('test');
});

test('matches snapshot', () => {
	const tree = renderer
		.create(<DarkButton label='test' disabled full />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
