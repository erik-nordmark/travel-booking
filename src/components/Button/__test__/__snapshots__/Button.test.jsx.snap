// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="button dark"
  data-testid="button"
  disabled={true}
  style={
    Object {
      "width": "100%",
    }
  }
  type="button"
>
  test
</button>
`;
